src/osm_lua_processing.cpp

```diff
@@ -344,7 +344,7 @@ void OsmLuaProcessing::Layer(const string &layerName, bool area) {
 				mp.push_back(p);
 			}
 
-            CorrectGeometry(mp);
+            if(!CorrectGeometry(mp)) return;
```

Comment:
I remove now polygons on input if they have self-intersection. Possibly, this is not a good idea. They might get rendered after simplification ?